### PR TITLE
Support GeneratedNameMaxLength property

### PR DESCRIPTION
**Pull request by regadas**
Addresses part of flyteorg/flyte#858. needs rebase after https://github.com/flyteorg/flyteplugins/pull/167

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/858

### CODE REVIEW

File: pkg/controller/nodes/task/taskexec_context.go [hunk 1]
```diff
@@ -136,7 +136,15 @@ func (t *Handler) newTaskExecutionContext(ctx context.Context, nCtx handler.Node
 		}
 	}
 
-	uniqueID, err := utils.FixedLengthUniqueIDForParts(IDMaxLength, nCtx.NodeExecutionMetadata().GetOwnerID().Name, currentNodeUniqueID, strconv.Itoa(int(id.RetryAttempt)))
+	length := IDMaxLength
+	if l := plugin.GetProperties().GeneratedNameMaxLength; l != nil {
+		length = *l
+		if length <= 0 {
+			return nil, errors.Errorf(errors.RuntimeExecutionError, nCtx.NodeID(), "GeneratedNameMaxLength shoud greater then 0")
```
Comment (EngHabu) [inline pkg/controller/nodes/task/taskexec_context.go:143]: I think this should fail earlier.. maybe at plug-in registration/initialization...

File: pkg/controller/nodes/task/k8s/plugin_manager.go [hunk 2]
```diff
@@ -109,18 +107,21 @@ func (e *PluginManager) AddObjectMetadata(taskCtx pluginsCore.TaskExecutionMetad
...
-	if cfg.InjectFinalizer && !e.disableInjectFinalizer {
+	if cfg.InjectFinalizer && !e.plugin.GetProperties().DisableInjectFinalizer {
 		f := append(o.GetFinalizers(), finalizer)
 		o.SetFinalizers(f)
 	}
 }
 
 func (e *PluginManager) GetProperties() pluginsCore.PluginProperties {
-	return pluginsCore.PluginProperties{}
+	props := e.plugin.GetProperties()
```
Comment (kumare3) [inline pkg/controller/nodes/task/k8s/plugin_manager.go:121]: why not just return e.plugin.GetProperties()?

Comment (regadas) [reply pkg/controller/nodes/task/k8s/plugin_manager.go:121]: right, so in this case `e.plugin` is a `k8s.Plugin` and `GetProperties` here returns a `k8s.PluginProperties` which can contain properties that only make sense for `k8s`.

**Review by EngHabu — APPROVED** [review]

File: pkg/controller/nodes/task/k8s/plugin_manager.go (hunk 2 above)
Comment (kumare3) [reply pkg/controller/nodes/task/k8s/plugin_manager.go:121]: aah oops, sorry missed that

**Review by kumare3 — APPROVED** [review]

**Pull request merged**